client/src/commands/createSmartContractProjectCommand.ts

```diff
@@ -92,22 +117,49 @@ export async function createFabricProject(): Promise<void> {
-        console.log('new fabric project folder is :' + folderSelect[0].fsPath);
+        console.log('new smart contract project folder is :' + folderSelect[0].fsPath);
         await commands.executeCommand('vscode.openFolder', folderSelect[0], true);
 
         // Run yo:fabric with default options in folderSelect
         // redirect to stdout as yo fabric prints to stderr
-        const yoFabricCmd: string = `yo fabric:chaincode -- --language=javascript --name="new-smart-contract" --version=0.0.1 --description="My Smart Contract" --author="John Doe" --license=Apache-2.0 2>&1`;
+        const yoFabricCmd: string = `yo fabric:chaincode -- --language="${chaincodeLanguage}" --name="new-smart-contract" --version=0.0.1 --description="My Smart Contract" --author="John Doe" --license=Apache-2.0 2>&1`;
         try {
             const yoFabricOut = await CommandUtil.sendCommand(yoFabricCmd, folderSelect[0].fsPath);
             outputAdapter.log(yoFabricOut);
-            outputAdapter.log('Successfully generated fabric project');
+            outputAdapter.log('Successfully generated smart contract project');
         } catch (error) {
-            console.log('found issue running yo:fabric command, see stderr:', error.stderr);
-            window.showErrorMessage('Issue creating fabric project');
+            console.log('found issue running yo:fabric command:', error);
+            window.showErrorMessage('Issue creating smart contract project');
             outputAdapter.log(error);
         }
 
     } // end of if folderSelect
 
-} // end of createFabricProject function
+} // end of createSmartContractProject function
+
+export async function getChaincodeLanguageOptions(): Promise<string[]> {
+    const yoFabricChild: child_process.ChildProcess = await child_process.spawn('/bin/sh', ['-c', 'yo fabric:chaincode < /dev/null']);
+    return new Promise<string[]>((resolve, reject) => {
+        yoFabricChild.on('exit', (returnCode: number) => {
+            const stdout: Buffer = yoFabricChild.stdout.read();
+            if (returnCode) {
+                return reject(new Error(`yo fabric: chaincode failed to run with return code ${returnCode}`));
+            } else if (stdout) {
+                const chaincodeLanguageArray: string[] = stdout.toString().split('\n');
+                chaincodeLanguageArray.shift();
+                const cleanChaincodeLangaugeArray: string[] = [];
+                for (const language of chaincodeLanguageArray) {
+                    if (language !== '') {
+                        const regex: RegExp = /^[^\w]*([\w]+)/g;
+                        const regexMatchArray: RegExpMatchArray = regex.exec(language);
+                        cleanChaincodeLangaugeArray.push(regexMatchArray[1]);
+                    }
+                }
+                console.log('printing available chaincode languages from yo fabric:chaincode output:', cleanChaincodeLangaugeArray);
+                resolve(cleanChaincodeLangaugeArray);
```

Comment:
There's a return for both reject cases so probably worth returning here as well. (Don't think it would cause problems with the current code but could do in the future?)